packages/eslint-config-checkout/index.js

```diff
@@ -1,30 +1,75 @@
 module.exports = {
   parser: "@typescript-eslint/parser",
+  plugins: ["@typescript-eslint", "formatjs"],
+  extends: [
+    "next",
+    "next/core-web-vitals",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "prettier",
+  ],
   parserOptions: {
-    tsconfigRootDir: "./",
-    project: ["./tsconfig.json"],
+    project: ["tsconfig.json"],
   },
-  plugins: ["@typescript-eslint"],
-  extends: ["prettier"],
   settings: {
     next: {
       rootDir: ["apps/*/", "packages/*/"],
     },
+    react: {
+      version: "999.999.999",
+    },
   },
   rules: {
+    "@next/next/no-html-link-for-pages": "off",
+    "react/jsx-key": "off",
     "array-callback-return": "error",
     "no-alert": ["error"],
     "no-debugger": ["error"],
     eqeqeq: ["error", "always", { null: "ignore" }],
-    "require-await": ["error"],
-    "no-restricted-syntax": [
+    "@typescript-eslint/no-misused-promises": ["error"],
+    "@typescript-eslint/no-floating-promises": ["error"],
+
+    "@typescript-eslint/ban-types": [
       "error",
       {
-        selector: "ForInStatement",
-        message: "for ... in disallowed, use for ... of instead",
+        types: {
+          // allow {} even though it's unsafe but comes handy
+          "{}": false,
+        },
       },
     ],
-    "@typescript-eslint/no-misused-promises": ["error"],
-    "@typescript-eslint/no-floating-promises": ["error"],
+
+    // we allow empty interfaces
+    "no-empty-pattern": "off",
+    "@typescript-eslint/no-empty-interface": "off",
+
+    // we allow empty functions
+    "@typescript-eslint/no-empty-function": "off",
+
+    // we sometimes use async functions that don't await anything
+    "@typescript-eslint/require-await": "off",
```

Comment:
Offf... I really don't like this. We've specifically added this rule due to issues in API routes in Next.js - sometimes forgetting `await` caused unexpected issues on production that were not visible in development.

I know that this can be annoying (you need to add either `void` or `await`) but IMHO this is a worthy tradeoff. Since API routes are main use case, we could enable it only for them? 🤔

Reply:
I think you're talking about another rule – `no-misused-promises` or `no-floating-promises` maybe?

`require-await` is actually about ensuring you only add `async` to functions that actually use `await` and remove `async` when functions don't use `await`. Which, in my opinion, is pointless – we often want `async` functions which don't use `await` – we just want to make sure they return a Promise. I don't see any value in this rule apart from **maybe** very minor (negligible) performance hit for using `async` where sync function would be enough.

Reply:
I'll also enable [`"@typescript-eslint/return-await": ["error", "in-try-catch"],`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/return-await.md)